internal/services/network/virtual_network_peering_resource_test.go

```diff
@@ -0,0 +1,218 @@
+	if err != nil {
+		return nil, err
+	}
+
+	future, err := client.Network.VnetPeeringsClient.Delete(ctx, id.ResourceGroup, id.VirtualNetworkName, id.Name)
+	if err != nil {
+		return nil, fmt.Errorf("deleting on virtual network peering: %+v", err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Network.VnetPeeringsClient.Client); err != nil {
+		return nil, fmt.Errorf("waiting for deletion of %s: %+v", *id, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (VirtualNetworkPeeringResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurestack" {
+  features {}
+}
+resource "azurestack_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+resource "azurestack_virtual_network" "test1" {
+  name                = "acctestvirtnet-1-%d"
+  resource_group_name = azurestack_resource_group.test.name
+  address_space       = ["10.0.1.0/24"]
+  location            = azurestack_resource_group.test.location
+}
+resource "azurestack_virtual_network" "test2" {
+  name                = "acctestvirtnet-2-%d"
+  resource_group_name = azurestack_resource_group.test.name
+  address_space       = ["10.0.2.0/24"]
+  location            = azurestack_resource_group.test.location
+}
+resource "azurestack_virtual_network_peering" "test1" {
+  name                         = "acctestpeer-1-%d"
+  resource_group_name          = azurestack_resource_group.test.name
+  virtual_network_name         = azurestack_virtual_network.test1.name
+  remote_virtual_network_id    = azurestack_virtual_network.test2.id
+  allow_virtual_network_access = true
+}
+resource "azurestack_virtual_network_peering" "test2" {
+  name                         = "acctestpeer-2-%d"
+  resource_group_name          = azurestack_resource_group.test.name
+  virtual_network_name         = azurestack_virtual_network.test2.name
+  remote_virtual_network_id    = azurestack_virtual_network.test1.id
+  allow_virtual_network_access = true
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.RandomInteger)
+}
+
+func (r VirtualNetworkPeeringResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+resource "azurestack_virtual_network_peering" "import" {
+  name                         = azurestack_virtual_network_peering.test1.name
+  resource_group_name          = azurestack_virtual_network_peering.test1.resource_group_name
+  virtual_network_name         = azurestack_virtual_network_peering.test1.virtual_network_name
+  remote_virtual_network_id    = azurestack_virtual_network_peering.test1.remote_virtual_network_id
+  allow_virtual_network_access = azurestack_virtual_network_peering.test1.allow_virtual_network_access
+}
+`, r.basic(data))
+}
+
+func (VirtualNetworkPeeringResource) basicUpdate(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurestack" {
+  features {}
+}
+resource "azurestack_resource_group" "test" {
```

Comment:
could we put new lines in here for readability?
```suggestion
}
resource "azurestack_resource_group" "test" {
```
```suggestion
}

resource "azurestack_resource_group" "test" {
```